.github/workflows/build_and_unitest.yml

```diff
@@ -9,9 +9,103 @@ on:
       - main
 
 jobs:
+  get_changed_modules:
+    runs-on: ubuntu-latest
+    name: Get changed modules
+
+    outputs: 
+      modified_modules: ${{ steps.output-changes.outputs.modified_modules }}
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # OR "2" -> To retrieve the preceding commit.
```

Comment:
lets make it two? How do things change?

Reply:
This was just some comments from the tj-actions example code I forgot to remove. But fetch-depth I think just compares the current commit in PR to the most recent or previous 1 or previous 2 (depends on what fetch depth is) commits in `main` but there is no reason to do so in our case?